File: CPP/546A.cpp
#include <iostream>

using namespace std;

int main () {
  size_t k, n, w;
  cin >> k >> n >> w;

  size_t total_cost = 0;
  for(int i = w; i > 0; --i)  total_cost += k * i;

  int diff = total_cost - n;
  cout << (diff < 0 ? 0 : diff) << endl;
}

File: Haskell/546A.hs
main :: IO ()
main = interact $ show . solve . map read . words

solve :: [Int] -> Int
solve [k, n, w] = if diff < 0 then 0 else diff
  where 
    totalCost k 0 = 0
    totalCost k w = k * w + totalCost k (w - 1)
    
    diff = totalCost k w - n